x-pack/plugins/apm/public/components/shared/unified_search_bar/index.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { Query, TimeRange } from '@kbn/es-query';
+import { useHistory, useLocation } from 'react-router-dom';
+import deepEqual from 'fast-deep-equal';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { EuiSkeletonRectangle } from '@elastic/eui';
+import qs from 'query-string';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import { TimePickerTimeDefaults } from '../date_picker/typings';
+import { ApmPluginStartDeps } from '../../../plugin';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmDataView } from '../../../hooks/use_apm_data_view';
+import { useProcessorEvent } from '../kuery_bar/use_processor_event';
```

Comment:
nit: Since the `useProcessorEvent` hook is used for both unified and kueryBar wouldn't fit better under the /hooks folders?

Reply:
it's a TODO on my list (Sticky Note on my Screen).
Since we are keeping the KueryBar for now i thought of referencing it directly from its folder. I would have moved it to Unified Search folder when we would have deleted it.

But i feel your point is valid, i will move it to `/hooks` folder so i won't have to worry about it while deleting KueryBar code. 👍🏼